### PR TITLE
java overridable_java_home_env

### PR DESCRIPTION
**Pull request by xu-cheng**
Closes #37622

### CODE REVIEW

File: Library/Homebrew/language/java.rb [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
-      { :JAVA_HOME => "$(/usr/libexec/java_home#{version_flag})" }
+      "/usr/libexec/java_home#{version_flag}"
+    end
+
+    def self.java_home_env(version=nil)
+      { :JAVA_HOME => "$(#{java_home_cmd(version)})" }
+    end
+
+    def self.overridable_java_home_env(version=nil)
+      { :JAVA_HOME => "$(if [ -z \"$JAVA_HOME\" ]; then echo \"$(#{java_home_cmd(version)})\"; else echo \"$JAVA_HOME\"; fi)" }
```
Comment (mikemcquaid) [inline Library/Homebrew/language/java.rb]: Maybe use `EOS.undent` and split this into multiple lines to make it easier to follow?

Comment (bfontaine) [inline Library/Homebrew/language/java.rb]: Shorter way:

```
{ :JAVA_HOME => "echo ${JAVA_HOME:-$(#{java_home_cmd(version)})}" }
```

In Bash, `${VAR:-default}` is `$VAR` if `VAR` exists, and `default` if not.

Comment (mikemcquaid) [inline Library/Homebrew/language/java.rb]: Yeh, that works for me (as long as the script is definitely using bash)

Comment (xu-cheng) [inline Library/Homebrew/language/java.rb]: Thanks @bfontaine  

@mikemcquaid The script is definitely using bash. c.f. https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/extend/pathname.rb#L376-L377

**Comment by xu-cheng** [comment]
Any more comment before I merging this?

**Comment by mikemcquaid** [comment]
:+1:

**Pull request closed without merging**